databases/databases.go

```diff
@@ -29,6 +29,15 @@ import (
 	"github.com/kbase/dts/databases/kbase"
 )
 
+// This error type is returned when a database is sought but not found.
+type NotFoundError struct {
+	dbName string
+}
+
+func (e NotFoundError) Error() string {
+	return fmt.Sprintf("The database '%s' was not found.", e.dbName)
+}
+
```

Comment:
Golang lesson time: what is the reason for having this error/error function here which is very similar to the DatabaseNotFoundError in the `core/database.go` package? In other languages you might have a package/file with a collection of different error types and import and use those errors where needed.

Reply:
Sorry--this is me experimenting with error types in an "early days" fashion. I haven't really used Go's error types before, since one can get by with string-flavored errors in relatively simple flows. I'd love to discuss this and other topics with you when you get some time to focus on CE/DTS stuff.

Speaking of which, I'm actually planning on removing the `core` package in another organization/cleanup PR, since it contains only interfaces that are used by other packages. Originally it was its own thing because the task manager lived there, but now that we have a `tasks` package, all the stuff in `core` can be moved to `databases`, `endpoints`, etc.